pages/k8s/charm-kubernetes-master.md

```diff
@@ -0,0 +1,629 @@
+
+The Kubernetes project defines some new terms that may be unfamiliar to users
+or operators. For more information please refer to the concept guide in the
+[getting started guide](https://kubernetes.io/docs/home/).
+
+This charm is an encapsulation of the Kubernetes master processes and the
+operations to run on any cloud for the entire lifecycle of the cluster.
+
+This charm is built from other charm layers using the Juju reactive framework.
+The other layers focus on specific subset of operations making this layer
+specific to operations of Kubernetes master processes.
+
+# Deployment
+
+This charm is not fully functional when deployed by itself. It requires other
+charms to model a complete Kubernetes cluster. A Kubernetes cluster needs a
+distributed key value store such as [Etcd](https://coreos.com/etcd/) and the
+kubernetes-worker charm which delivers the Kubernetes node services. A cluster
+requires a Software Defined Network (SDN), a Container Runtime such as
+[containerd](https://jaas.ai/u/containers/containerd), and Transport Layer
+Security (TLS) so the components in a cluster communicate securely.
+
+Please take a look at the [Canonical Distribution of Kubernetes](https://jujucharms.com/canonical-kubernetes/)
+or the [Kubernetes core](https://jujucharms.com/kubernetes-core/) bundles for
+examples of complete models of Kubernetes clusters.
+
+# Resources
+
+The kubernetes-master charm takes advantage of the [Juju Resources](https://jujucharms.com/docs/2.0/developer-resources)
+feature to deliver the Kubernetes software.
+
+In deployments on public clouds the Charm Store provides the resource to the
+charm automatically with no user intervention. Some environments with strict
+firewall rules may not be able to contact the Charm Store. In these network
+restricted  environments the resource can be uploaded to the model by the Juju
+operator.
+
+#### Snap Refresh
+
+The kubernetes resources used by this charm are snap packages. When not
+specified during deployment, these resources come from the public store. By
+default, the `snapd` daemon will refresh all snaps installed from the store
+four (4) times per day. A charm configuration option is provided for operators
+to control this refresh frequency.
+
+>NOTE: this is a global configuration option and will affect the refresh
+time for all snaps installed on a system.
+
+Examples:
+
+```sh
+## refresh kubernetes-master snaps every tuesday
+juju config kubernetes-master snapd_refresh="tue"
+
+## refresh snaps at 11pm on the last (5th) friday of the month
+juju config kubernetes-master snapd_refresh="fri5,23:00"
+
+## delay the refresh as long as possible
+juju config kubernetes-master snapd_refresh="max"
+
+## use the system default refresh timer
+juju config kubernetes-master snapd_refresh=""
+```
+
+For more information on the possible values for `snapd_refresh`, see the
+*refresh.timer* section in the [system options][] documentation.
```

Comment:
Let's change this to something like: "For more information, see the /kubernetes/docs/snap-refresh documentation."